### PR TITLE
[ALLUXIO-2075] Consolidate stripPrefixIfPresent mothed into Util

### PR DESCRIPTION
**Pull request by xgong**
https://alluxio.atlassian.net/browse/ALLUXIO-2075

### CODE REVIEW

**Comment by xgong** [comment]
alluxio-bot, check this please

**Comment by aaudiber** [comment]
alluxio-bot, check this please

File: underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java [hunk 1]
```diff
@@ -653,12 +654,7 @@ private String stripFolderSuffixIfPresent(String key) {
...
   private String stripPrefixIfPresent(String key) {
-    if (key.startsWith(mBucketPrefix)) {
-      return key.substring(mBucketPrefix.length());
-    }
-    if (key.startsWith(PATH_SEPARATOR)) {
-      return key.substring(PATH_SEPARATOR.length());
-    }
-    return key;
+    key = CommonUtils.stripPrefixIfPresent(key, mBucketPrefix);
+    return CommonUtils.stripPrefixIfPresent(key, PATH_SEPARATOR);
```
Comment (calvinjia) [inline underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java]: This does not provide the same behavior as before, ie. `prefix:///file` will be parsed differently.

Comment (xgong) [inline underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java]: Oh, my mistake. You are right

**Comment by calvinjia** [comment]
@xgong This looks good overall, could you also apply the changes to the similar code in `S3AUnderFileSystem`?

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**